diff --git a/dist/docgen/docgen-worker.js b/dist/docgen/docgen-worker.js
index b4b15a4462f8af2fbd09a2135e4f3c60d1bfc406..f6d0715f55f070a47bf027cd1f1373fae6e2800b 100644
--- a/dist/docgen/docgen-worker.js
+++ b/dist/docgen/docgen-worker.js
@@ -81,7 +81,7 @@ async function buildDocgenPayload(input, context) {
 var createDocgenProvider = () => createLazyDocgenMiddleware({
   createManager: async () => {
     try {
-      let ts = await import("typescript");
+      let ts = await import("typescript-docgen");
       return new ComponentMetaManager(ts);
     } catch {
       return;
diff --git a/dist/preset.js b/dist/preset.js
index 59f3aab6d7e35b520b5fc4c89789721121cfeb16..6a52e0fef39172b28d5719be4e8a77e6f9603a29 100644
--- a/dist/preset.js
+++ b/dist/preset.js
@@ -298,7 +298,7 @@ var REACT_NODE_TYPES = /* @__PURE__ */ new Set([
 ]);
 var getTsConfig = async (componentFilePath) => {
   try {
-    let ts = await import("typescript"), tsconfigPath = componentFilePath ? findTsconfigPathForFile(dirname2(componentFilePath), componentFilePath) : ts.findConfigFile(process.cwd(), ts.sys.fileExists);
+    let ts = await import("typescript-docgen"), tsconfigPath = componentFilePath ? findTsconfigPathForFile(dirname2(componentFilePath), componentFilePath) : ts.findConfigFile(process.cwd(), ts.sys.fileExists);
     if (tsconfigPath === void 0)
       return {};
     let basePath = dirname2(tsconfigPath), { config, error } = ts.readConfigFile(tsconfigPath, ts.sys.readFile);
@@ -362,7 +362,7 @@ var componentMetaManagerPromise;
 function getSharedComponentMetaManager() {
   return componentMetaManagerPromise || (componentMetaManagerPromise = (async () => {
     try {
-      let ts = await import("typescript");
+      let ts = await import("typescript-docgen");
       return new ComponentMetaManager(ts);
     } catch {
       logger3.debug(
