diff --git a/dist/core-server/index.js b/dist/core-server/index.js
index 8ac44a995f5cf06c1199adf2dc2afec0423a4a74..5724edfae3beee8024b533c554fbf01136494a39 100644
--- a/dist/core-server/index.js
+++ b/dist/core-server/index.js
@@ -5687,7 +5687,10 @@ async function extractStorybookMetadata(outputFile, configDir) {
 function useStorybookMetadata(app, configDir) {
   app.use("/project.json", async (req, res) => {
     let storybookMetadata = await getStorybookMetadata(configDir);
-    res.setHeader("Content-Type", "application/json"), res.write(JSON.stringify(storybookMetadata)), res.end();
+    res.setHeader("Content-Type", "application/json"), res.setHeader("Access-Control-Allow-Origin", "*"), res.setHeader(
+      "Access-Control-Allow-Headers",
+      "Origin, X-Requested-With, Content-Type, Accept"
+    ), res.write(JSON.stringify(storybookMetadata)), res.end();
   });
 }
 
